segarray with empty segments and `nan`s

The fix from https://github.com/Bears-R-Us/arkouda/pull/3233 doesn't seem to handle empty segments and `nan`s (despite this working locally). I'll dig into this more but want the fix for ints to be merged asap since it was a user request

```
FAILED PROTO_tests/tests/io_test.py::TestParquet::test_empty_segs_segarray - AssertionError: Series are different

Series values are different (37.1505 %)
[index]: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, ...]
[left]:  [[], [], [], [nan], [], [-826.8812045081526, nan], [746.5395385504996], [nan], [900.3690478055171], [], [nan, -759.1574320740729], [], [nan, 970.6743505032746], [nan, -670.4402397767861, nan], [125.09200179134109, nan, 635.5648194386292], [nan], [], [], [-505.549499783878, nan], [396.7222631500233], [], [], [nan], [906.8508500586117], [nan], [419.64192221289954], [nan], [], [-885.7667312239257], [nan, 889.6722720385171], [], [nan], [], [], [-657.44636913257, nan, -199.85141274364685], [], [], [], [nan], [], [987.1184929236938], [nan, -816.8662079575722, nan, -782.2443499566617, nan], [-464.5106205446067, nan], [], [689.4230808363716, nan], [-371.64280367283175], [nan], [], [757.0918420165312, nan], [], [693.4700619455009, nan], [], [-235.64453401995445], [nan], [68.344077077262], [nan], [-138.677430177774], [], [nan, 784.1140793634186], [], [nan, -511.5728860896983], [], [nan], [-68.8965684581591], [nan, -60.48939392606576, nan], [823.314[563](https://github.com/Bears-R-Us/arkouda/actions/runs/9213085553/job/25346161805?pr=3233#step:7:564)6027548], [nan], [-979.599409315703], [nan], [], [], [-731.0089274212505], [nan], [], [], [], [522.6579703899522], [nan], [], [525.2058629395156], [nan, -294.7196584213302], [nan, -708.710500788178, nan], [-166.59934577693957, nan], [], [], [313.0480718484919], [nan, 735.0177573450717, nan, 171.4041798442438, nan, -325.47902976321893], [nan, -79.95589074638644], [nan, 667.5804860426608], [nan], [], [304.59112235744715], [], [], [nan, 403.40421040373644, nan, -965.9072537116001], [nan], [], [345.3145581042936], [nan], [214.61226846222075], ...]
[right]: [[], [], [], [nan], [], [-826.8812045081526, nan], [746.5395385504996], [nan], [900.3690478055171], [], [nan, -759.1574320740729], [], [nan, 970.6743505032746], [nan, -670.4402397767861, nan], [125.09200179134109, nan, 635.[564](https://github.com/Bears-R-Us/arkouda/actions/runs/9213085553/job/25346161805?pr=3233#step:7:565)8194386292], [nan], [], [], [-505.549499783878, nan], [396.7222631500233], [], [], [nan], [906.8508500586117], [nan], [419.64192221289954], [nan], [], [-885.7667312239257], [nan, 889.6722720385171], [], [nan], [], [], [-657.44636913257, nan, -199.85141274364685], [], [], [], [nan], [], [987.1184929236938], [nan, -816.8662079575722, nan, -782.2443499566617, nan], [-464.5106205446067, nan], [], [689.4230808363716, nan], [-371.64280367283175], [nan], [], [757.0918420165312, nan], [], [693.4700619455009, nan], [], [-235.64453401995445], [nan], [68.344077077262], [nan], [-138.677430177774], [], [nan, 784.1140793634186], [], [nan, -511.5728860896983], [], [nan], [-68.8965684581591], [nan, -60.48939392606576, nan], [823.3145636027548], [nan], [-979.599409315703], [nan], [], [], [-731.0089274212505], [nan], [], [], [], [522.6579703899522], [nan], [], [525.2058629395156], [nan, -294.7196584213302], [nan, -708.710500788178, nan], [-166.59934577693957, nan], [], [], [313.0480718484919], [nan, 735.0177573450717, nan, 171.4041798442438, nan, -325.47902976321893], [nan, -79.95589074638644], [nan, 667.5804860426608], [nan], [], [304.59112235744715], [], [], [nan, 403.40421040373644, nan, -965.9072537116001], [nan], [], [345.3145581042936], [nan], [214.61226846222075], ...]
At positional index 262817, first diff: [30.17138893715355] != [-510.38933605702505]
====== 1 failed, 1369 passed, 2 skipped, 35 warnings in 453.24s (0:07:33) ======
```

once this is fixed we can reenable by adding `some_nans` back in

```python
some_nans = rng.uniform(-(2**10), 2**10, val_size)
some_nans[ak.arange(val_size) % 2 == 0] = np.nan
vals_list = [
    ...,
    some_nans,
]
```

--- a/PROTO_tests/tests/io_test.py
+++ b/PROTO_tests/tests/io_test.py
@@ -537,7 +537,7 @@ class TestParquet:
         df_dict = dict()
         seed = np.random.default_rng().choice(2**63)
         rng = ak.random.default_rng(seed)
-        some_nans = rng.uniform(-(2 ** 10), 2 ** 10, val_size)
+        some_nans = rng.uniform(-(2**10), 2**10, val_size)
         some_nans[ak.arange(val_size) % 2 == 0] = np.nan
         vals_list = [
             rng.uniform(-(2**10), 2**10, val_size),
@@ -551,7 +551,9 @@ class TestParquet:
             # segs must start with 0, all other segment lengths are random
             # by having val_size number of segments, except in the extremely unlikely case of
             # randomly getting exactly arange(val_size), we are guaranteed empty segs
-            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))])
+            segs = ak.concatenate(
+                [ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))]
+            )
             df_dict["rand"] = ak.SegArray(segs, vals).to_list()
 
             pddf = pd.DataFrame(df_dict)
@@ -567,7 +569,7 @@ class TestParquet:
                 # we pass the same absolute and relative tolerances as the numpy default in allclose
                 # to ensure float point differences don't cause errors
                 print("\nseed: ", seed)
-                assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
+                assert_series_equal(pddf["rand"], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
 
 class TestHDF5:

--- a/PROTO_tests/tests/io_test.py
+++ b/PROTO_tests/tests/io_test.py
@@ -551,7 +551,7 @@ class TestParquet:
             # segs must start with 0, all other segment lengths are random
             # by having val_size number of segments, except in the extremely unlikely case of
             # randomly getting exactly arange(val_size), we are guaranteed empty segs
-            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1))])
+            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))])
             df_dict["rand"] = ak.SegArray(segs, vals).to_list()
 
             pddf = pd.DataFrame(df_dict)

--- a/PROTO_tests/tests/io_test.py
+++ b/PROTO_tests/tests/io_test.py
@@ -535,12 +535,16 @@ class TestParquet:
         val_size = 1000000
 
         df_dict = dict()
-        rng = ak.random.default_rng()
+        seed = np.random.default_rng().choice(2**63)
+        rng = ak.random.default_rng(seed)
+        some_nans = rng.uniform(-(2 ** 10), 2 ** 10, val_size)
+        some_nans[ak.arange(val_size) % 2 == 0] = np.nan
         vals_list = [
             rng.uniform(-(2**10), 2**10, val_size),
             rng.integers(0, 2**32, size=val_size, dtype="uint"),
             rng.integers(0, 1, size=val_size, dtype="bool"),
             rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
+            some_nans,
         ]
 
         for vals in vals_list:
@@ -562,6 +566,7 @@ class TestParquet:
                 # are lists of lists. assert_series_equal handles this and properly handles nans.
                 # we pass the same absolute and relative tolerances as the numpy default in allclose
                 # to ensure float point differences don't cause errors
+                print("\nseed: ", seed)
                 assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
 

--- a/src/ArrowFunctions.cpp
+++ b/src/ArrowFunctions.cpp
@@ -585,20 +585,6 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
         int16_t definition_level; // needed for any type that is nullable
 
         std::shared_ptr<parquet::ColumnReader> column_reader = row_group_reader->Column(idx);
-
-
-
-
-
-
-
-
-
-
-
-
-
-
         if(lty == ARROWINT64 || lty == ARROWUINT64) {
           int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (int64_t*)chpl_arr;
@@ -608,38 +594,13 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
 
           while (reader->HasNext() && arrayIdx < numElems) {
             (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
-            // if values_read is 0, that means that it was a null value
+            // if values_read is 0, that means that it was an empty seg
             if (values_read != 0) {
               arrayIdx++;
             }
             i++;
           }
-        } 
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        
-        else if(lty == ARROWINT32 || lty == ARROWUINT32) {
+        } else if(lty == ARROWINT32 || lty == ARROWUINT32) {
           int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (int64_t*)chpl_arr;
           parquet::Int32Reader* reader =
@@ -649,7 +610,7 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
           int32_t tmp;
           while (reader->HasNext() && arrayIdx < numElems) {
             (void)reader->ReadBatch(1, &definition_level, nullptr, &tmp, &values_read);
-            // if values_read is 0, that means that it was a null value
+            // if values_read is 0, that means that it was an empty seg
             if (values_read != 0) {
               chpl_ptr[arrayIdx] = (int64_t)tmp;
               arrayIdx++;
@@ -683,65 +644,29 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
 
           while (reader->HasNext() && arrayIdx < numElems) {
             (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
-            // if values_read is 0, that means that it was a null value
+            // if values_read is 0, that means that it was an empty seg
             if (values_read != 0) {
               arrayIdx++;
             }
             i++;
           }
         } else if(lty == ARROWFLOAT) {
+          // convert to simpler single batch to sidestep this seemingly architecture dependent (see issue #3234)
+          int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (double*)chpl_arr;
           parquet::FloatReader* reader =
             static_cast<parquet::FloatReader*>(column_reader.get());
-          
-          while (reader->HasNext() && i < numElems) {
-            if((numElems - i) < batchSize) // adjust batchSize if needed
-              batchSize = numElems - i;
-            float* tmpArr = new float[batchSize]; // this will not include NaN values
-            int16_t* def_lvl = new int16_t[batchSize];
-            int16_t* rep_lvl = new int16_t[batchSize];
-            (void)reader->ReadBatch(batchSize, def_lvl, rep_lvl, tmpArr, &values_read);
 
-            int64_t tmp_offset = 0; // used to properly access tmp after NaN encountered
-            int64_t val_idx = 0;
-            for (int64_t j = 0; j< batchSize; j++){
-              // skip any empty segments
-              if (def_lvl[j] == 1)
-                continue;
-              
-              // Null values treated as NaN
-              if (def_lvl[j] == 2) {
-                chpl_ptr[i+val_idx] = NAN;
-                tmp_offset++; // adjustment for values array since Nulls are not included
-              }
-              else if (def_lvl[j] == 3){ // defined value to write
-                chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
-              }
-              val_idx++;
-            }
-
-            i += values_read + tmp_offset; // account for values and NaNs, but not empty segments
-
-            delete[] tmpArr;
-            delete[] def_lvl;
-            delete[] rep_lvl;
-          }
-        } else if(lty == ARROWDOUBLE) {
-          std::cout << "\n\n\n\n\n\n\n\n we are double\n";
-
-          int16_t definition_level; // nullable type and only reading single records in batch
-          auto chpl_ptr = (double*)chpl_arr;
-          parquet::DoubleReader* reader =
-            static_cast<parquet::DoubleReader*>(column_reader.get());
-
+          float tmp;
           while (reader->HasNext() && arrayIdx < numElems) {
-            (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
-            // if values_read is 0, that means that it was a null value
+            (void)reader->ReadBatch(1, &definition_level, nullptr, &tmp, &values_read);
+            // if values_read is 0, that means that it was a null value or empty seg
             if (values_read != 0) {
+              chpl_ptr[arrayIdx] = (double) tmp;
               arrayIdx++;
             }
             else {
-              // check if nan otherwise just treat as empty seg
+              // check if nan otherwise it's an empty seg
               if (definition_level == 2) {
                 chpl_ptr[arrayIdx] = NAN;
                 arrayIdx++;
@@ -749,41 +674,29 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
             }
             i++;
           }
-        } 
+        } else if(lty == ARROWDOUBLE) {
+          // convert to simpler single batch to sidestep this seemingly architecture dependent (see issue #3234)
+          int16_t definition_level; // nullable type and only reading single records in batch
+          auto chpl_ptr = (double*)chpl_arr;
+          parquet::DoubleReader* reader =
+            static_cast<parquet::DoubleReader*>(column_reader.get());
 
-          // while (reader->HasNext() && i < numElems) {
-          //   if((numElems - i) < batchSize) // adjust batchSize if needed
-          //     batchSize = numElems - i;
-          //   double* tmpArr = new double[batchSize]; // NaNs not included here
-          //   int16_t* def_lvl = new int16_t[batchSize];
-          //   int16_t* rep_lvl = new int16_t[batchSize];
-          //   (void)reader->ReadBatch(batchSize, def_lvl, rep_lvl, tmpArr, &values_read);
-            
-          //   int64_t tmp_offset = 0; // used to properly access tmp after NaN encountered
-          //   int64_t val_idx = 0;
-          //   for (int64_t j = 0; j< batchSize; j++){
-          //     // skip any empty segments
-          //     if (def_lvl[j] == 1)
-          //       continue;
-              
-          //     // Null values treated as NaN
-          //     if (def_lvl[j] == 2) {
-          //       chpl_ptr[i+val_idx] = NAN;
-          //       tmp_offset++; // adjustment for values array since Nulls are not included
-          //     }
-          //     else if (def_lvl[j] == 3){ // defined value to write
-          //       chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
-          //     }
-          //     val_idx++;
-          //   }
-
-          //   i += values_read + tmp_offset; // account for values and NaNs, but not empty segments
-
-          //   delete[] tmpArr;
-          //   delete[] def_lvl;
-          //   delete[] rep_lvl;
-          // }
-        // }
+          while (reader->HasNext() && arrayIdx < numElems) {
+            (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
+            // if values_read is 0, that means that it was a null value or empty seg
+            if (values_read != 0) {
+              arrayIdx++;
+            }
+            else {
+              // check if nan otherwise it's an empty seg
+              if (definition_level == 2) {
+                chpl_ptr[arrayIdx] = NAN;
+                arrayIdx++;
+              }
+            }
+            i++;
+          }
+        }
       }
       return 0;
     }

--- a/src/ArrowFunctions.cpp
+++ b/src/ArrowFunctions.cpp
@@ -585,6 +585,20 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
         int16_t definition_level; // needed for any type that is nullable
 
         std::shared_ptr<parquet::ColumnReader> column_reader = row_group_reader->Column(idx);
+
+
+
+
+
+
+
+
+
+
+
+
+
+
         if(lty == ARROWINT64 || lty == ARROWUINT64) {
           int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (int64_t*)chpl_arr;
@@ -600,7 +614,32 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
             }
             i++;
           }
-        } else if(lty == ARROWINT32 || lty == ARROWUINT32) {
+        } 
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        
+        else if(lty == ARROWINT32 || lty == ARROWUINT32) {
           int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (int64_t*)chpl_arr;
           parquet::Int32Reader* reader =
@@ -688,43 +727,63 @@ int cpp_readListColumnByName(const char* filename, void* chpl_arr, const char* c
             delete[] rep_lvl;
           }
         } else if(lty == ARROWDOUBLE) {
+          std::cout << "\n\n\n\n\n\n\n\n we are double\n";
+
+          int16_t definition_level; // nullable type and only reading single records in batch
           auto chpl_ptr = (double*)chpl_arr;
           parquet::DoubleReader* reader =
             static_cast<parquet::DoubleReader*>(column_reader.get());
 
-          while (reader->HasNext() && i < numElems) {
-            if((numElems - i) < batchSize) // adjust batchSize if needed
-              batchSize = numElems - i;
-            double* tmpArr = new double[batchSize]; // NaNs not included here
-            int16_t* def_lvl = new int16_t[batchSize];
-            int16_t* rep_lvl = new int16_t[batchSize];
-            (void)reader->ReadBatch(batchSize, def_lvl, rep_lvl, tmpArr, &values_read);
-            
-            int64_t tmp_offset = 0; // used to properly access tmp after NaN encountered
-            int64_t val_idx = 0;
-            for (int64_t j = 0; j< batchSize; j++){
-              // skip any empty segments
-              if (def_lvl[j] == 1)
-                continue;
-              
-              // Null values treated as NaN
-              if (def_lvl[j] == 2) {
-                chpl_ptr[i+val_idx] = NAN;
-                tmp_offset++; // adjustment for values array since Nulls are not included
-              }
-              else if (def_lvl[j] == 3){ // defined value to write
-                chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
-              }
-              val_idx++;
+          while (reader->HasNext() && arrayIdx < numElems) {
+            (void)reader->ReadBatch(1, &definition_level, nullptr, &chpl_ptr[arrayIdx], &values_read);
+            // if values_read is 0, that means that it was a null value
+            if (values_read != 0) {
+              arrayIdx++;
             }
-
-            i += values_read + tmp_offset; // account for values and NaNs, but not empty segments
-
-            delete[] tmpArr;
-            delete[] def_lvl;
-            delete[] rep_lvl;
+            else {
+              // check if nan otherwise just treat as empty seg
+              if (definition_level == 2) {
+                chpl_ptr[arrayIdx] = NAN;
+                arrayIdx++;
+              }
+            }
+            i++;
           }
-        }
+        } 
+
+          // while (reader->HasNext() && i < numElems) {
+          //   if((numElems - i) < batchSize) // adjust batchSize if needed
+          //     batchSize = numElems - i;
+          //   double* tmpArr = new double[batchSize]; // NaNs not included here
+          //   int16_t* def_lvl = new int16_t[batchSize];
+          //   int16_t* rep_lvl = new int16_t[batchSize];
+          //   (void)reader->ReadBatch(batchSize, def_lvl, rep_lvl, tmpArr, &values_read);
+            
+          //   int64_t tmp_offset = 0; // used to properly access tmp after NaN encountered
+          //   int64_t val_idx = 0;
+          //   for (int64_t j = 0; j< batchSize; j++){
+          //     // skip any empty segments
+          //     if (def_lvl[j] == 1)
+          //       continue;
+              
+          //     // Null values treated as NaN
+          //     if (def_lvl[j] == 2) {
+          //       chpl_ptr[i+val_idx] = NAN;
+          //       tmp_offset++; // adjustment for values array since Nulls are not included
+          //     }
+          //     else if (def_lvl[j] == 3){ // defined value to write
+          //       chpl_ptr[i+val_idx] = (double)tmpArr[val_idx-tmp_offset];
+          //     }
+          //     val_idx++;
+          //   }
+
+          //   i += values_read + tmp_offset; // account for values and NaNs, but not empty segments
+
+          //   delete[] tmpArr;
+          //   delete[] def_lvl;
+          //   delete[] rep_lvl;
+          // }
+        // }
       }
       return 0;
     }

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -614,8 +614,10 @@ class ParquetTest(ArkoudaTest):
                 file_path = f"{tmp_dirname}/empty_segs"
                 pddf.to_parquet(file_path)
                 akdf = ak.DataFrame(ak.read_parquet(file_path))
+                pddupe = pd.DataFrame(pd.read_parquet(file_path))
 
                 to_pd = pd.Series(akdf["rand"].values.to_list())
+                pddupe_series = pd.Series(pddupe["rand"].values)
                 # raises an error if the two series aren't equal
                 # we can't use np.allclose(pddf['rand'].to_list, akdf['rand'].to_list) since these
                 # are lists of lists. assert_series_equal handles this and properly handles nans.
@@ -628,6 +630,11 @@ class ParquetTest(ArkoudaTest):
                 print("pddf['rand'][263047]: ", pddf['rand'][263047])
                 print("akdf['rand'][263047]: ", akdf['rand'][263047])
                 print("to_pd[263047]: ", to_pd[263047])
+                print()
+                print("pddupe['rand'][263047]: ", pddupe['rand'][263047])
+                print("pddupe_series[263047]: ", pddupe_series[263047])
+                assert_series_equal(pddf['rand'], pddupe_series, check_names=False, rtol=1e-05, atol=1e-08)
+                assert_series_equal(pddf['rand'], pddupe['rand'], check_names=False, rtol=1e-05, atol=1e-08)
                 assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
     @pytest.mark.optional_parquet

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -589,15 +589,16 @@ class ParquetTest(ArkoudaTest):
         val_size = 1000000
 
         df_dict = dict()
-        seed = np.random.default_rng().choice(2**63)
+        # seed = np.random.default_rng().choice(2**63)
+        seed = 9219146137719780068
         rng = ak.random.default_rng(seed)
         some_nans = rng.uniform(-(2 ** 10), 2 ** 10, val_size)
         some_nans[ak.arange(val_size) % 2 == 0] = np.nan
         vals_list = [
-            rng.uniform(-(2**10), 2**10, val_size),
-            rng.integers(0, 2**32, size=val_size, dtype="uint"),
-            rng.integers(0, 1, size=val_size, dtype="bool"),
-            rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
+            # rng.uniform(-(2**10), 2**10, val_size),
+            # rng.integers(0, 2**32, size=val_size, dtype="uint"),
+            # rng.integers(0, 1, size=val_size, dtype="bool"),
+            # rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
             some_nans,
         ]
 
@@ -605,7 +606,7 @@ class ParquetTest(ArkoudaTest):
             # segs must start with 0, all other segment lengths are random
             # by having val_size number of segments, except in the extremely unlikely case of
             # randomly getting exactly arange(val_size), we are guaranteed empty segs
-            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1))])
+            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))])
             df_dict["rand"] = ak.SegArray(segs, vals).to_list()
 
             pddf = pd.DataFrame(df_dict)

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -626,6 +626,7 @@ class ParquetTest(ArkoudaTest):
                 print("pd version: ", pd.__version__)
                 print()
                 print("pddf['rand'][263047]: ", pddf['rand'][263047])
+                print("akdf['rand'][263047]: ", akdf['rand'][263047])
                 print("to_pd[263047]: ", to_pd[263047])
                 assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -622,6 +622,11 @@ class ParquetTest(ArkoudaTest):
                 # we pass the same absolute and relative tolerances as the numpy default in allclose
                 # to ensure float point differences don't cause errors
                 print("\nseed: ", seed)
+                print("np version: ", np.__version__)
+                print("pd version: ", pd.__version__)
+                print()
+                print("pddf['rand'][263047]: ", pddf['rand'][263047])
+                print("to_pd[263047]: ", to_pd[263047])
                 assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
     @pytest.mark.optional_parquet

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -589,12 +589,16 @@ class ParquetTest(ArkoudaTest):
         val_size = 1000000
 
         df_dict = dict()
-        rng = ak.random.default_rng()
+        seed = np.random.default_rng().choice(2**63)
+        rng = ak.random.default_rng(seed)
+        some_nans = rng.uniform(-(2 ** 10), 2 ** 10, val_size)
+        some_nans[ak.arange(val_size) % 2 == 0] = np.nan
         vals_list = [
             rng.uniform(-(2**10), 2**10, val_size),
             rng.integers(0, 2**32, size=val_size, dtype="uint"),
             rng.integers(0, 1, size=val_size, dtype="bool"),
             rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
+            some_nans,
         ]
 
         for vals in vals_list:
@@ -616,6 +620,7 @@ class ParquetTest(ArkoudaTest):
                 # are lists of lists. assert_series_equal handles this and properly handles nans.
                 # we pass the same absolute and relative tolerances as the numpy default in allclose
                 # to ensure float point differences don't cause errors
+                print("\nseed: ", seed)
                 assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
     @pytest.mark.optional_parquet

--- a/tests/parquet_test.py
+++ b/tests/parquet_test.py
@@ -586,19 +586,21 @@ class ParquetTest(ArkoudaTest):
 
     def test_empty_segs_segarray(self):
         # verify reproducer for #3074 is resolved
+
+        # bug seemed to consistently appear for val_sizes
+        # exceeding 700000, round up to ensure we'd hit it
         val_size = 1000000
 
         df_dict = dict()
-        # seed = np.random.default_rng().choice(2**63)
-        seed = 9219146137719780068
+        seed = np.random.default_rng().choice(2**63)
         rng = ak.random.default_rng(seed)
-        some_nans = rng.uniform(-(2 ** 10), 2 ** 10, val_size)
+        some_nans = rng.uniform(-(2**10), 2**10, val_size)
         some_nans[ak.arange(val_size) % 2 == 0] = np.nan
         vals_list = [
-            # rng.uniform(-(2**10), 2**10, val_size),
-            # rng.integers(0, 2**32, size=val_size, dtype="uint"),
-            # rng.integers(0, 1, size=val_size, dtype="bool"),
-            # rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
+            rng.uniform(-(2**10), 2**10, val_size),
+            rng.integers(0, 2**32, size=val_size, dtype="uint"),
+            rng.integers(0, 1, size=val_size, dtype="bool"),
+            rng.integers(-(2**32), 2**32, size=val_size, dtype="int"),
             some_nans,
         ]
 
@@ -606,7 +608,9 @@ class ParquetTest(ArkoudaTest):
             # segs must start with 0, all other segment lengths are random
             # by having val_size number of segments, except in the extremely unlikely case of
             # randomly getting exactly arange(val_size), we are guaranteed empty segs
-            segs = ak.concatenate([ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))])
+            segs = ak.concatenate(
+                [ak.array([0]), ak.sort(ak.randint(0, val_size, val_size - 1, seed=seed))]
+            )
             df_dict["rand"] = ak.SegArray(segs, vals).to_list()
 
             pddf = pd.DataFrame(df_dict)
@@ -614,28 +618,15 @@ class ParquetTest(ArkoudaTest):
                 file_path = f"{tmp_dirname}/empty_segs"
                 pddf.to_parquet(file_path)
                 akdf = ak.DataFrame(ak.read_parquet(file_path))
-                pddupe = pd.DataFrame(pd.read_parquet(file_path))
 
                 to_pd = pd.Series(akdf["rand"].values.to_list())
-                pddupe_series = pd.Series(pddupe["rand"].values)
                 # raises an error if the two series aren't equal
                 # we can't use np.allclose(pddf['rand'].to_list, akdf['rand'].to_list) since these
                 # are lists of lists. assert_series_equal handles this and properly handles nans.
                 # we pass the same absolute and relative tolerances as the numpy default in allclose
                 # to ensure float point differences don't cause errors
                 print("\nseed: ", seed)
-                print("np version: ", np.__version__)
-                print("pd version: ", pd.__version__)
-                print()
-                print("pddf['rand'][263047]: ", pddf['rand'][263047])
-                print("akdf['rand'][263047]: ", akdf['rand'][263047])
-                print("to_pd[263047]: ", to_pd[263047])
-                print()
-                print("pddupe['rand'][263047]: ", pddupe['rand'][263047])
-                print("pddupe_series[263047]: ", pddupe_series[263047])
-                assert_series_equal(pddf['rand'], pddupe_series, check_names=False, rtol=1e-05, atol=1e-08)
-                assert_series_equal(pddf['rand'], pddupe['rand'], check_names=False, rtol=1e-05, atol=1e-08)
-                assert_series_equal(pddf['rand'], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
+                assert_series_equal(pddf["rand"], to_pd, check_names=False, rtol=1e-05, atol=1e-08)
 
     @pytest.mark.optional_parquet
     def test_against_standard_files(self):

